core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -314,7 +316,7 @@ private[spark] class ExecutorAllocationManager(
       // Do not change our target while we are still initializing,
       // Otherwise the first job may have to ramp up unnecessarily
       0
-    } else if (maxNeeded < numExecutorsTarget) {
+    } else if (maxNeeded <= numExecutorsTarget) {
       // The target number exceeds the number we actually need, so stop adding new
```

Comment:
yeah I don't think this change hurts, but unless you have a reason for changing it, I'd prefer to leave it just for when we need to dig through history.